metrics-influxdb/src/main/java/io/dropwizard/metrics5/influxdb/InfluxDbReporter.java

```diff
@@ -0,0 +1,517 @@
+        if (nameChars != null) {
+            str.append(nameChars);
+        } else {
+            writeMeasurementNoCache(name);
+            nameChars = new char[str.length()];
+            str.getChars(0, str.length(), nameChars, 0);
+            encodedNameCache.put(name, nameChars);
+        }
+        
+        str.append(' ');
+        firstField = true;
+    }
+    
+    private void writeMeasurementNoCache(MetricName name) {
+        MetricName prefixedName = MetricName.join(prefix, name);
+        
+        appendName(prefixedName.getKey(), str);
+        Map<String, String> tags = prefixedName.getTags();
+        if (!tags.isEmpty()) {
+            // InfluxDB Performance and Setup Tips:
+            // Sort tags by key before sending them to the database. 
+            // The sort should match the results from the Go bytes.Compare function.
+            TreeMap<String, String> sortedTags = new TreeMap<>(tags);
+            for (Map.Entry<String, String> tag : sortedTags.entrySet()) {
+                str.append(',');
+                appendName(tag.getKey(), str);
+                str.append('=');
+                appendName(tag.getValue(), str);
+            }
+        }
+    }
+    
+    private void writeTimestampMillis(long utcMillis) {
+        str.append(' ').append(utcMillis).append("000000\n");
+    }
+    
+    private boolean hasValues() {
+        return !firstField;
+    }
+    
+    private void reportLine() throws IOException {
+        sender.send(str);
+    }
+
+    private void writeFieldIfEnabled(MetricAttribute key, double value) {
+        if (!getDisabledMetricAttributes().contains(key)) {
+            writeField(key);
+            writeFieldValue(value);
+        }
+    }
+    
+    private void writeFieldIfEnabled(MetricAttribute key, long value) {
+        if (!getDisabledMetricAttributes().contains(key)) {
+            writeField(key);
+            writeFieldValue(value);
+        }
+    }
+    
+    private void writeField(MetricAttribute key) {
+        if (!firstField) {
+            str.append(',');
+        }
+        str.append(key.getCode()) // no need to escape, only safe chars
+           .append('=');
+        firstField = false;
+    }
+    
+    private void writeField(String key) {
+        if (!firstField) {
+            str.append(',');
+        }
+        appendName(key, str);
+        str.append('=');
+        firstField = false;
+    }
+    
+    private void writeFieldValue(double value) {
+        str.append(value);
+    }
+    
+    private void writeFieldValue(long value) {
+        str.append(value).append('i');
+    }
+    
+    private void writeFieldValue(String value) {
+        str.append('"');
+        appendString(value, str);
+        str.append('"');
+    }
+    
+    private void writeFieldValue(boolean value) {
+        str.append(value ? 't' : 'f');
+    }
+    
+    private static void appendName(CharSequence field, StringBuilder dst) {
+        int len = field.length();
```

Comment:
Looks like that the `len` variable is redundant. We could directly use `field.length` in the loop condition.